scripts/DeployFraxOFTProtocol.s.sol

```diff
@@ -0,0 +1,132 @@
+                FraxOFTUpgradeable.initialize.selector,
+                "Frax Share",
+                "FXS",
+                delegate
+            )
+        });
+
+        // Deploy sFRAX
+        address sFraxOft = deployFraxOFTUpgradeableAndProxy({
+            _oftBytecode: type(OFTUpgradeable).creationCode,
+            _constructorArgs: abi.encode(endpoint),
+            _initializeArgs: abi.encodeWithSelector(
+                FraxOFTUpgradeable.initialize.selector,
+                "Staked Frax",
+                "sFRAX",
+                delegate
+            )
+        });
+
+        // sfrxETH
+        address frxEthOft = deployFraxOFTUpgradeableAndProxy({
+            _oftBytecode: type(OFTUpgradeable).creationCode,
+            _constructorArgs: abi.encode(endpoint),
+            _initializeArgs: abi.encodeWithSelector(
+                FraxOFTUpgradeable.initialize.selector,
+                "Frax Ether",
+                "frxETH",
+                delegate
+            )
+        });
+
+        // Deploy FRAX
+        address sFraxOft = deployFraxOFTUpgradeableAndProxy({
+            _oftBytecode: type(OFTUpgradeable).creationCode,
+            _constructorArgs: abi.encode(endpoint),
+            _initializeArgs: abi.encodeWithSelector(
+                FraxOFTUpgradeable.initialize.selector,
+                "Frax",
+                "FRAX",
+                delegate
+            )
+        });
+
+        // Deploy FPI
+        // address frxEthOft = deployFraxOFTUpgradeableAndProxy({
+        //     _oftBytecode: type(OFTUpgradeable).creationCode,
+        //     _constructorArgs: abi.encode(endpoint),
+        //     _initializeArgs: abi.encodeWithSelector(
+        //         FraxOFTUpgradeable.initialize.selector,
+        //         "Frax Price Index",
+        //         "FPI",
+        //         delegate
+        //     )
+        // });
+
+    // TODO: set peers, set enforced options, set delegate, transfer owership of oft, transfer proxy ownership
+
+    }
+
+
+    /// @notice Sourced from https://github.com/FraxFinance/LayerZero-v2-upgradeable/blob/e1470197e0cffe0d89dd9c776762c8fdcfc1e160/oapp/test/TestHelper.sol#L266
+    function deployFraxOFTUpgradeableAndProxy(
+        bytes memory _oftBytecode,
+        bytes memory _constructorArgs,
+        bytes memory _initializeArgs,
+        address _delegate
+    ) returns (address implementation, address proxy) {
+        bytes memory bytecode = bytes.concat(abi.encodePacked(_oftBytecode), _constructorArgs);
+        vm.broadcast(basicDeployer);
+        assembly {
+            implementation := create(0, add(bytecode, 0x20), mload(bytecode))
+            if iszero(extcodesize(implementation)) {
+                revert(0, 0)
+            }
+        }
+        /// @dev: OFT delegate is proxy admin
+        vm.broadcast(proxyDeployer);
+        proxy = address(new TransparentUpgradeableProxy(implementation, basicDeployer, _initializeArgs));
```

Comment:
Nit: proxy admin will be the pk used to deploy

Reply:
wdym?  I don't think the proxy admin contract can deploy proxies...

My original thought is we stick with using msig over `ProxyAdmin` as the same ownership/upgrade can happen within one batch tx, and because the contract is not actually tested in the audited repo:
https://github.com/FraxFinance/LayerZero-v2-upgradeable/blob/e1470197e0cffe0d89dd9c776762c8fdcfc1e160/oapp/test/TestHelper.sol#L58

Reply:
sorry the proxy terminology is confusing 😵‍💫

> proxy admin 
as in the admin of the proxy not the tertiary ProxyAdmin contract. 

And yeah makes sense only thing to be aware of is that if we have function selector clashing between proxy/Implementation the admin will not be able to access, but this is easy to solve.

Reply:
Would you personally lay the `ProxyAdmin` on top or no?  My current take is to stick with msig to stay aligned with tests and assume the implementation is never intended to be called.

Reply:
Personally I would but totally up to you one of those grey area things. 

No issue w/ the current config as long as the proxy admin owner (the msig) is not going to interact w/ the proxy contract outside of proxy maintenance

Reply:
Ah I see.  We are intending to interact with the proxy as we'll need to set peers, delegate, etc so it may be worth adding `ProxyAdmin`.  We can discuss on team call today find consensus.

Reply:
Word, only really an issue if there is a `changeAdmin(address)` or other function selector clash.

Reply:
There isn't an issue with the current spec but I'm thinking it's best to maintain the standard recommended by OZ.